src/main/java/com/entropyteam/entropay/employees/clients/services/ClientService.java

```diff
@@ -0,0 +1,60 @@
+package com.entropyteam.entropay.employees.clients.services;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.UUID;
+import org.springframework.data.domain.Sort;
+import org.springframework.stereotype.Service;
+import com.entropyteam.entropay.employees.clients.dtos.ClientDto;
+import com.entropyteam.entropay.employees.clients.dtos.ClientSaveRequestDto;
+import com.entropyteam.entropay.employees.clients.dtos.ClientSaveResponseDto;
+import com.entropyteam.entropay.employees.clients.models.Client;
+import com.entropyteam.entropay.employees.clients.repositories.ClientRepository;
+import com.entropyteam.entropay.employees.common.exceptions.ResourceNotFoundException;
+import com.entropyteam.entropay.employees.common.mappers.ClientMapper;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+
+@Service
+@Log4j2
+@RequiredArgsConstructor
+public class ClientService {
+
+    private final ClientRepository clientRepository;
+
+    public ClientSaveResponseDto createClient(ClientSaveRequestDto newClientDto) {
+        Client newClient = ClientMapper.MAPPER.toEntity(newClientDto);
+        newClient.setActive(true);
+        return ClientMapper.MAPPER.toSaveResponseDto(clientRepository.save(newClient));
+    }
+
+    public ClientDto findById(String clientId) {
+        return ClientMapper.MAPPER.toDto(clientRepository.findByIdAndIsActive(UUID.fromString(clientId), true)
+                .orElseThrow(() -> new ResourceNotFoundException("Client does not exist: " + clientId)));
+    }
+
+    public ClientSaveResponseDto updateClient(String clientId, ClientSaveRequestDto clientDto) {
+        Client client = clientRepository.findByIdAndIsActive(UUID.fromString(clientId), true)
+                .orElseThrow(() -> new ResourceNotFoundException("Client does not exist: " + clientId));
+        client.setAddress(clientDto.getAddress());
+        client.setContact(clientDto.getContact());
+        client.setPreferredCurrency(clientDto.getPreferredCurrency());
+        client.setModifiedOn(LocalDateTime.now());
```

Comment:
I think this will done by the annotation on the entity.

Reply:
We can even use mapper's toEntity method but AFAIK It will replace the reference of this entity.